Procfile

```diff
@@ -1,4 +1,4 @@
 web: ./start.sh
-jobs: bundle exec bin/delayed_job run
+jobs: bundle exec bin/delayed_job --pool=Default:2 --pool=NSIDC,LPDAAC:2 --pool=*:2 run
```

Comment:
I dont see any reason this wouldnt work (from an NGAP pserspective), but there are a few ways to do this which are worth considering:
- In the approach you have now, 6 delayed_job workers will be running in the same ECS task (AKA in its own docker container).  I do not know what the memory/cpu implications of adding a worker are, and we can scale up RAM/CPU if we need to, but the point is that the workers will be sharing resources
- You could also scale up the number of 'jobs' tasks running in NGAP, meaning that n separate docker containers will be running the same delayed_job command (which does not seem to be what you want since you seem to have different types of queues).
- Finally, you could take sort of a hybrid approach.  basically, create multiple jobs tasks in the Procfile with unique names (e.g. nsidc_jobs, lpdaac_jobs...) and have each of those run some subset of the workers.  This way, you have container separation, but each is doing something different.  Unless you are confident that there will be no resource contention in the current approach, I would suggest trying this.  

As always, try this out in searchlab or sit before merging to make sure everything works as expected.

Reply:
Totally agree with testing on SIT - it's definitely required here.

It is true that the additional workers will take up additional memory, however I can't fathom it's all that much.  As you said as well - memory is also pretty cheap and can be added.  From all the documentation and chatter, using multiple workers as well as queues with delayed_job is rather common.

Reply:
Tim -- I'm wary of the additional complexity, though I see the merit of your hybrid approach.  

How about we start simple -- i.e. all workers running in the same ESC task / docker container -- see how that works, and reach for the hybrid solution if we have to?

Reply:
That is fine with me, just throwing out the options :-) as far as the memory/CPU load, it may be hard to determine that in a non-prod env, so make sure you/ngap are keeping an eye on it when it goes to prod so we know if we need to bump something up.

Reply:
Tim -  In the light of what we're just now learning about EDSC restarts, ooms and load capacity, before we deploy this I'd like to re-ask your opinion here.  What is the safest way  allocate workers in the EDSC cluster?  Knowing what we know now, it unwise to run all workers in the same container?

Reply:
@tgoff, can you please respond to @andyfekete's comment here? We need to get this deployed if everything checks out. And we need to come up with an alternative if there's more to be done. Either way, this is a _huge_ priority for NSIDC, so we need to close the loop quickly.

Reply:
I think we could go either way.  separate containers means more isolation, if one queue goes down or is overwhelmed, it does not affect the others.  However, there is some overhead associated with an additional docker container.  we have like 4 extra GB of RAM to work with (though it is possible we can go a bit higher depending on how deployments are done).  We could easily spin up 2 more jobs tasks for those queues if we want.  Ultimately that probably is the most resilient solution

Reply:
@tgoff, I'm not sure I follow 100%. Are you saying it's okay to move forward with things as is and we bump up RAM on the NGAP side? Or something else? I don't know how all of the specifics work, but I just want a clear "yes" or "no" from someone that what we are proposing is okay or not.

Reply:
So two jobs tasks total, per the hybrid approach you mentioned above (ie - one for hi priority daacs, and the other for everybody else)?

Reply:
well, we actually have 2 jobs currently: 
```
=== jobs
Desired: 2
Running: 2
RAM: 1024
CPU: 256
```
we could make one of those high priority and one normal, or add a third if current capacity is not sufficient.  TO do this, you would replace the single `jobs` line in the Procfile with two lines e.g. `normal_jobs` and `priority_jobs`.

Reply:
yeah, that is all (with each line specifically calling out the queues to work).  Then we can work on making sure the jobs are properly scaled and resourced.  But the more I think about it, the more it seems smart to separate the queues into different instances

Reply:
@mreese84, @tgoff, just to be explicit, is this what we're aiming for now? 

normal_jobs: bundle exec bin/delayed_job --pool=Default run
priority_jobs: bundle exec bin/delayed_job --pool=NSIDC,LPDAAC run

Realize that's only a single worker for 'priority', so if there is a long running task in there, I think it may 'bottleneck' following priority jobs.

Reply:
that looks good.  you can also set priority to 2 runners (and increase memory if needed) and/or scale priority to 2 tasks

Reply:
Wait -- so if we go to two tasks, we lose the capability of multiple workers per queue?

Reply:
@andyfekete - not at all!  I would think this would be better:

normal_jobs: bundle exec bin/delayed_job --pool=Default:2 run
priority_jobs: bundle exec bin/delayed_job --pool=NSIDC,LPDAAC:2 run

This would allow two workers per task/queue set.  Would this work for everyone?  Let me know and I'll update the PR if so.

Reply:
Works for me if it works for @andyfekete

Reply:
looks good.  we will need to watch RAM and make sure we scale it accordingly.  is there any way to produce an ops like load in sit or UAT?

Reply:
John, in this scheme is it possible to dedicate one worker to each priority daac? That way one couldn't starve the other.

Reply:
@andyfekete - I believe this should do it:

normal_jobs: bundle exec bin/delayed_job --pool=Default:2 run
priority_jobs: bundle exec bin/delayed_job --pool=NSIDC --pool=LPDAAC run

Procfile updated and pushed up.  :)

Reply:
Looks good.  Engage!

Reply:
@andyfekete @mreese84 excellent - I'll merge here shortly :)